### DISCUSSION TITLE
How do I import an external library to my tests

### DISCUSSION THREAD

**Discussion by Onokaev** [microsoft/typespec#2699] [Q&A] [answered]
I have tests that depend on a typespec library that is on npm. This typespec library contains types that I am using in my tests.
How do I let the test runner know about these tests?

I have tried adding the package to my devDependencies then adding it to the tests using createTestWrapper method like so:
`createTestWrapper(host, {
    autoUsings: ["TypeSpec.Http"],
    wrapper: (code) => `${prefix} ${code}`,
    compilerOptions: {
      miscOptions: { "disable-linter": true },
    },
    autoImports: ["typespec-lib1"]
  });`

All my tests fail when I add this library autoImport

**Comment by timotheeguerin** [comment, marked as answer]
Hi @Onokaev, you can find docs on how to setup your test environment included how to include another library in your test host here

https://microsoft.github.io/typespec/extending-typespec/basics#testing

The test host needs to know ahead of time what are the files that are going to be needed(tsp, js, etc.) which is why you have to register those test libraries.

Example here on how to include `@typespec/rest` with the `RestTestLibrary`

```
import { createTestHost, createTestWrapper } from "@typespec/compiler/testing";
import { HttpTestLibrary } from "@typespec/http/testing";
import { MyTestLibrary } from "../src/testing/index.js";

export async function createMyTestHost() {
  return createTestHost({
    libraries: [HttpestLibrary, MyTestLibrary], // Add other libraries you depend on in your tests
  });
}
export async function createMyTestRunner() {
  const host = await createOpenAPITestHost();
  return createTestWrapper(host, { autoUsings: ["Typespec.Http"] });
}
```
